### PR TITLE
Fix merge form block strategy

### PR DESCRIPTION
**Pull request by alavieille**
[OO-BUGFIX] Fix merge form block strategy with transformer and subscriber
[OO-BCBREAK] Constructor of class  ``Backoffice/EventSubscriber/BlockTypeSubscriber`` is changed
[OO-BCBREAK] Constructor of class ``Backoffice/Form/Type/BlockType`` is changed

### CODE REVIEW

File: Backoffice/EventSubscriber/BlockTypeSubscriber.php [hunk 1]
```diff
@@ -3,90 +3,56 @@
...
-     * @param array                $fixedParameters
-     * @param FormFactoryInterface $formFactory
-     * @param int                  $blockPosition
+     * @param array $fixedParameters
      */
-    public function __construct(GenerateFormManager $generateFormManager, $fixedParameters, FormFactoryInterface $formFactory, $blockPosition = 0)
+    public function __construct(array $fixedParameters)
     {
-        $this->generateFormManager = $generateFormManager;
         $this->fixedParameters = $fixedParameters;
```
Comment (ngilain) [inline Backoffice/EventSubscriber/BlockTypeSubscriber.php]: @alavieille : Can we check the preconditions on the array or not ?

File: Backoffice/Tests/Form/Type/BlockTypeTest.php [hunk 2]
```diff
@@ -28,13 +28,13 @@ public function setUp()
...
         $blockToArrayTransformer = Phake::mock('OpenOrchestra\Backoffice\Form\DataTransformer\BlockToArrayTransformer');
+        $blockTypeSubscriber = Phake::mock('OpenOrchestra\Backoffice\EventSubscriber\BlockTypeSubscriber');
 
         Phake::when($this->generateFormManager)->getTemplate(Phake::anyParameters())->thenReturn($this->templateName);
         $this->fixedParameters = array('component', 'submit', 'label', 'class', 'id', 'max_age');
         $this->formFactory = Phake::mock('Symfony\Component\Form\FormFactoryInterface');
 
-        $this->blockType = new BlockType($this->generateFormManager, $this->fixedParameters, $this->formFactory, $blockToArrayTransformer);
-    }
+        $this->blockType = new BlockType($this->generateFormManager, $blockToArrayTransformer, $blockTypeSubscriber);    }
```
Comment (ngilain) [inline Backoffice/Tests/Form/Type/BlockTypeTest.php]: } feels alone at the end of the line :)

File: BackofficeBundle/Resources/config/subscriber.yml [hunk 3]
```diff
@@ -137,3 +138,9 @@ services:
             - @open_orchestra_model.repository.node
         tags:
             - { name: kernel.event_subscriber }
+    open_orchestra_backoffice.subscriber.block_form_type:
+         class: %open_orchestra_backoffice.subscriber.block_form_type.class%
+         arguments:
+             - %open_orchestra_backoffice.block.fixed_attributes%
+         tags:
+             - { name: kernel.event_subscriber }
```
Comment (ngilain) [inline BackofficeBundle/Resources/config/subscriber.yml]: NL@EOF

**Comment by alavieille** [comment]
@ngilain pr updated

**Comment by itkg-nanne** [comment]
:+1:

**Comment by ngilain** [comment]
:+1:

**Pull request merged**